internal/commands/accesslist/update_test.go

```diff
@@ -0,0 +1,222 @@
+						},
+						AllowedIPsFn: func(groupID, appID string) ([]realm.AllowedIP, error) {
+							return allowedIPs, nil
+						},
+					}
+				},
+				expectedErr: errors.New("something went wrong with the app"),
+			},
+			{
+				description: "if there is an issue with finding allowed ips for the app",
+				clientSetup: func() realm.Client {
+					return mock.RealmClient{
+						FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+							return []realm.App{app}, nil
+						},
+						AllowedIPsFn: func(groupID, appID string) ([]realm.AllowedIP, error) {
+							return nil, errors.New("something happened with allowed IPs")
+						},
+					}
+				},
+				expectedErr: errors.New("something happened with allowed IPs"),
+			},
+			{
+				description: "if there is an issue with finding the allowed ip specified in the access list",
+				inputs:      updateInputs{Address: "1.1.1.1"},
+				clientSetup: func() realm.Client {
+					return mock.RealmClient{
+						FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+							return []realm.App{app}, nil
+						},
+						AllowedIPsFn: func(groupID, appID string) ([]realm.AllowedIP, error) {
+							return allowedIPs, nil
+						},
+					}
+				},
+				expectedErr: errors.New("unable to find allowed IP: 1.1.1.1"),
+			},
+			{
+				description: "if there is an issue with updating the allowed ip",
+				inputs:      updateInputs{Address: allowedIPs[0].Address},
+				clientSetup: func() realm.Client {
+					return mock.RealmClient{
+						FindAppsFn: func(filter realm.AppFilter) ([]realm.App, error) {
+							return []realm.App{app}, nil
+						},
+						AllowedIPsFn: func(groupID, appID string) ([]realm.AllowedIP, error) {
+							return allowedIPs, nil
+						},
+						AllowedIPUpdateFn: func(groupID, appID, allowedIPID, newAddress, comment string) error {
+							return errors.New("something bad happened")
+						},
+					}
+				},
+				expectedErr: errors.New("something bad happened"),
+			},
+		} {
+			t.Run(tc.description, func(t *testing.T) {
+				_, ui := mock.NewUI()
+
+				realmClient := tc.clientSetup()
+				cmd := &CommandUpdate{tc.inputs}
+				assert.Equal(t, tc.expectedErr, cmd.Handler(nil, ui, cli.Clients{Realm: realmClient}))
+			})
+		}
+	})
+}
+
+func TestAllowedIPUpdateInputs(t *testing.T) {
+	allowedIPs := []realm.AllowedIP{
+		{ID: "address1", Address: "0.0.0.0", Comment: "comment"},
+		{ID: "address2", Address: "192.1.1.1"},
+		{ID: "address3", Address: "192.158.1.38", Comment: "cool comment"},
+	}
+
+	t.Run("resolving the allowed ip", func(t *testing.T) {
+		t.Run("should return the allowed ip if searching by address", func(t *testing.T) {
+			inputs := updateInputs{
+				Address: "0.0.0.0",
+			}
+			result, err := inputs.resolveAllowedIP(nil, allowedIPs)
+			assert.Nil(t, err)
+			assert.Equal(t, allowedIPs[0], result)
+		})
+
+		t.Run("should return an error if we cannot find the allowed ip to update", func(t *testing.T) {
```

Comment:
avoid words like "we"... something along the lines of:

`"should return an error if allowed ip is not found"`